n3fit/src/n3fit/hyper_optimization/mongofiletrials.py

```diff
@@ -0,0 +1,250 @@
+"""
+    Hyperopt trial object for parallel hyperoptimization with MongoDB.
+    Data are fetched from MongoDB databases and stored in the form of json files within the nnfit folder
+"""
+import json
+import logging
+import os
+import subprocess
+
+from bson import SON, ObjectId
+from hyperopt.mongoexp import MongoTrials
+import tensorflow as tf
+
+from n3fit.hyper_optimization.filetrials import space_eval_trial
+
+log = logging.getLogger(__name__)
+
+
+def get_physical_gpus():
+    """
+    Retrieve a list of all physical GPU devices available in the system.
+
+    Returns
+    -------
+        list: A list of TensorFlow physical devices of type 'GPU'.
+    """
+    return tf.config.list_physical_devices('GPU')
+
+
+def set_tf_visible_device(gpu_id, devices_list):
```

Comment:
Is this now a deprecated function?
```suggestion
```

Reply:
Thanks @Radonirinaunimi! I made it but I never used. I removed it completely. Also, should we move `get_physical_gpus` to the keras_backend like in `operations.py` ? Note that to make it work I am already importing tensorflow from `mongofiletrials.py`

Reply:
If this should be moved, actually, it might make more sense to put it in `keras_backend.internal_state` instead (?).

Reply:
Thanks! It seems more logic indeed. Done.